src/controllers/update/overseas.entity.review.controller.ts

```diff
@@ -15,6 +18,30 @@ export const get = (req: Request, res: Response, next: NextFunction) => {
     const changeLinkUrl: string = config.OVERSEAS_ENTITY_UPDATE_DETAILS_URL;
     const overseasEntityHeading: string = "Check the overseas entity details";
 
+    if (!isActiveFeature(config.FEATURE_FLAG_DISABLE_UPDATE_PRIVATE_DATA_FETCH)) {
+
+      // Fetch OE email address if not already in session.
+      const overseasEntityId = appData.overseas_entity_id;
+      const transactionId = appData.transaction_id;
+      if (!appData.entity?.email && overseasEntityId !== undefined && transactionId !== undefined) {
+        if (appData.entity === undefined) {
+          appData.entity = {};
+        }
+
+        const privateOeDetails = await getPrivateOeDetails(req, transactionId, overseasEntityId);
+        if (privateOeDetails === undefined || privateOeDetails.email_address === undefined || privateOeDetails.email_address.length === 0) {
+          const message = `Private OE Details not found for overseas entity ${appData.entity_number}`;
+          logger.error(message);
```

Comment:
There's a logger function we can use here which includes the request
```suggestion
          logger.errorRequest(req, message);
```